backend/src/schema/resolvers/reports.js

```diff
@@ -120,23 +119,24 @@ export default {
       const session = context.driver.session()
       const { id } = parent
       let filed
-      const readTxPromise = session.readTransaction(async tx => {
-        const cypher = `
-          MATCH (submitter:User)-[filed:FILED]->(report:Report {id: $id})
-          RETURN filed, submitter
-        `
-        const params = { id }
-        const allReportsTransactionResponse = await tx.run(cypher, params)
-        log(allReportsTransactionResponse)
-        return allReportsTransactionResponse.records.map(record => ({
+      const readTxPromise = session.readTransaction(async transaction => {
+        const filedReportsTransactionResponse = await transaction.run(
+          `
+            MATCH (submitter:User)-[filed:FILED]->(report:Report {id: $id})
+            RETURN filed, submitter
+          `,
+          { id },
+        )
+        log(filedReportsTransactionResponse)
+        return filedReportsTransactionResponse.records.map(record => ({
           submitter: record.get('submitter').properties,
           filed: record.get('filed').properties,
         }))
       })
       try {
-        const txResult = await readTxPromise
-        if (!txResult[0]) return null
-        filed = txResult.map(reportedRecord => {
+        const filedReports = await readTxPromise
+        if (!filedReports) return []
```

Comment:
Same, the promise is returning an array why this check?

Reply:
copy :spaghetti: